examples/chip-tool/commands/common/Command.cpp

```diff
@@ -171,7 +171,7 @@ bool HandleNullableOptional(Argument & arg, char * argValue, std::function<bool(
     if (arg.isNullable())
     {
         auto * nullable = reinterpret_cast<chip::app::DataModel::Nullable<T> *>(arg.value);
-        if (strcmp(argValue, "null") == 0)
+        if (argValue != nullptr && strncmp(argValue, "null", 4) == 0)
```

Comment:
How can `argValue` be not null-terminated? So how do we then know the size of it?

Reply:
Yes, it is impossible in this case, we use strncmp for argument comparisons in other cases, but in one instance, we use strcmp(argValue, "null").

strncmp would only be necessary if argValue was not null-terminated and had a length of exactly 4, making strcmp unsafe. However, since argValue is always null-terminated in this context, it is safe to use either strncmp(argValue, "null", 4) or strcmp(argValue, "null").

This change is intended to standardize the method used for argument comparison in the code.

Reply:
Just to follow up on the Slack thread about this: "standardizing" breaks the code.  The places that use strncmp are doing _prefix_ checks, but this one is supposed to be an _equality_ check.   We don't want to treat all strings starting with "null" as the null value.